src/psyclone/domain/lfric/lfric_extract_driver_creator.py

```diff
@@ -915,12 +865,40 @@ def create(self, nodes, input_list, output_list,
 
         return file_container
 
+    # -------------------------------------------------------------------------
+    @staticmethod
+    def collect_all_required_modules(file_container):
+        '''Collects recursively all modules used in the file container.
+        It returns a dictionary, with the keys being all the (directly or
+        indirectly used modules)
+
+        :param file_container:
+        :type file_container: \
+            :py:class:`psyclone.psyir.psyir.nodes.FileContainer`
+
+        :returns: a dictionary, with the required module names as key, and \
+            as value a set of all modules required by the key module.
+        :rtype: Set[str]
+
+        '''
+        sym_tab = file_container.children[0].symbol_table
+
+        # Add all modules imported, except intrinsic ones
+        all_mods = set()
+        for symbol in sym_tab.symbols:
+            if isinstance(symbol, ContainerSymbol) and \
+                    ",intrinsic" not in symbol.name:
+                all_mods.add(symbol.name)
+
+        mod_manager = ModuleManager.get()
+        return mod_manager.get_all_dependencies_recursively(all_mods)
+
     # -------------------------------------------------------------------------
     def get_driver_as_string(self, nodes, input_list, output_list,
                              prefix, postfix, region_name,
                              writer=FortranWriter()):
-        # pylint: disable=too-many-arguments
-        '''This function uses ``create()`` function to get a PSyIR of a
+        # pylint: disable=too-many-arguments, too-many-locals
```

Comment:
"...uses the `create()`..." and "the PSyIR" instead of "a PSyIR".

Reply:
Done.